great_expectations/core/batch_config.py

```diff
@@ -0,0 +1,14 @@
+from dataclasses import dataclass
+
+from great_expectations.datasource.fluent.interfaces import DataAsset
+
+
+@dataclass(frozen=True)
```

Comment:
We can do this in the future PR.

This should probably be a pydantic model since we will want to serialize/deserialize from config.
When we do this, we will need to think about serialization/deserialization since we likely want to reference the data asset by name in the config. Maybe our config model objects should live in the same directory?